src/sourmash/sketchcomparison.py

```diff
@@ -0,0 +1,138 @@
+        #if not (self.mh1_cmp.track_abundance and self.mh2_cmp.track_abundance):
+        #    return self.mh1_cmp.angular_similarity(self.mh2_cmp)
+        #else:
+        #    return ""
+
+    @property
+    def cosine_similarity(self):
+        return self.angular_similarity
+
+
+@dataclass
+class NumMinHashComparison(BaseMinHashComparison):
+    """Class for standard comparison between two scaled minhashes"""
+    cmp_num: int = None
+
+    def __post_init__(self):
+        "Initialize NumMinHashComparison using values from provided MinHashes"
+        if self.cmp_num is None: # record the num we're doing this compa#rison on
+            self.cmp_num = min(self.mh1.num, self.mh2.num)
+        self.check_comparison_compatibility()
+        self.downsample_and_handle_ignore_abundance(cmp_num=self.cmp_num)
+
+@dataclass
+class FracMinHashComparison(BaseMinHashComparison):
+    """Class for standard comparison between two scaled minhashes"""
+    cmp_scaled: int = None # scaled value for this comparison (defaults to maximum scaled between the two sigs)
+    threshold_bp: int = 0
+
+    def __post_init__(self):
+        "Initialize ScaledComparison using values from provided FracMinHashes"
+        if self.cmp_scaled is None: # record the scaled we're doing this comparison on
+            self.cmp_scaled = max(self.mh1.scaled, self.mh2.scaled)
+        self.check_comparison_compatibility()
+        self.downsample_and_handle_ignore_abundance(cmp_scaled=self.cmp_scaled)
+        # for these, do we want the originals, or the cmp_scaled versions?? (or both?). do we need them at all?
+        self.mh1_scaled = self.mh1.scaled
+        self.mh2_scaled = self.mh2.scaled
+
+    @property
+    def pass_threshold(self):
+        return self.intersect_bp >= self.threshold_bp
+
+    @property
+    def intersect_bp(self):
+        return len(self.intersect_mh) * self.cmp_scaled
+
+    @property
+    def mh1_containment(self):
+        return self.mh1_cmp.contained_by(self.mh2_cmp)
+
+    @property
+    def mh2_containment(self):
+        return self.mh2_cmp.contained_by(self.mh1_cmp)
+
+    @property
+    def max_containment(self):
+        return self.mh1_cmp.max_containment(self.mh2_cmp)
+
+    @property
+    def avg_containment(self):
+        return np.mean([self.mh1_containment, self.mh2_containment])
+
+    def weighted_intersection(self, from_mh=None, from_abundD={}):
+         # map abundances to all intersection hashes.
+        abund_mh = self.intersect_mh.copy_and_clear()
+        abund_mh.track_abundance = True
+        if from_mh is not None:
+            from_abundD = from_mh.hashes
+        if from_abundD is not None:
+            #abundD[k] for k in intersect_mh.hashes
+            abunds = {k: from_abundD.get(k, 1) for k in self.intersect_mh.hashes }
+            abund_mh.set_abundances(abunds)
+            return abund_mh
+        return self.intersect_mh # or do we want to set all abundances to 1?
```

Comment:
huh, the default return is not ever executed here?!

Reply:
should be covered now after https://github.com/sourmash-bio/sourmash/commit/4e06116c4d8a4a6bc573935d1ca019ac68e0773f

Reply:
...or not. I shall go fix

Reply:
actually, it's ok - was just showing an old codecov